connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -643,7 +643,7 @@ else if (value instanceof ByteBuffer)
                 }
                 case STRUCT: {
                     Struct struct = (Struct) value;
-                    if (struct.schema() != schema)
+                    if (!struct.schema().equals(schema))
```

Comment:
A few notes:
* It is possible for `schema` to be null
* `struct` should never be null, since all of the existing/current `LogicalTypeConverter` implementations in `TO_JSON_LOGICAL_CONVERTERS` always return a non-null value.
• `struct.schema()` should never be null, since it's not possible to create a `Struct` without a `Schema`.

So, this change should be good and won't result in an NPE as long as the above conditions are true.

Reply:
Yeah that's what I expected although I didn't analyze it since I didn't have time, was a quick fix for necessary for a project.